File: js/169_MajorityElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    if(nums.length < 2) return nums[0];
    nums.sort();
    return nums[Math.floor(nums.length/2)];
};


/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    var winner = 0, winnerCount = 0;
    for(var i in nums){
        if(winnerCount === 0){
            winner = nums[i];
            winnerCount++;
        }else if(winner == nums[i]){
            winnerCount++;
        }else{
            winnerCount--;
        }
    }
    return winner;
};

File: cpp/169_MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
    	std::sort(nums.begin(), nums.end());
        return nums[nums.size()/2];
    }
};

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int winner = 0;
        int winnerCount = 0;
        for(int i = 0; i < nums.size(); ++i){
            if(winnerCount == 0){
                winner = nums[i];
                winnerCount++;
            }else if(winner == nums[i]){
                winnerCount++;
            }else{
                winnerCount--;
            }
        }
        return winner;
    }
};